service/javadsl/integration-client/src/main/java/com/lightbend/lagom/javadsl/client/integration/LagomClientFactory.java

```diff
@@ -243,11 +253,20 @@ private void closeGracefully(Effect close) {
     private static class StaticServiceLocator extends CircuitBreakingServiceLocator {
         private final URI uri;
 
-        StaticServiceLocator(CircuitBreakers circuitBreakers, URI uri) {
-            super(circuitBreakers);
+        StaticServiceLocator(CircuitBreakersPanel circuitBreakersPanel, URI uri) {
+            super(circuitBreakersPanel);
             this.uri = uri;
         }
 
+        /**
+         * @deprecated Use constructor accepting {@link CircuitBreakersPanel} instead
+         */
+        @Deprecated
+        StaticServiceLocator(com.lightbend.lagom.internal.client.CircuitBreakers circuitBreakers, URI uri) {
+            this(CircuitBreakersConverter.toJavadslCircuitBreakerInvoker(circuitBreakers), uri);
+        }
```

Comment:
That's true. I didn't see that. 

Now I realised that there is some differences in Scala / Java API here. In Java these classes are private and we have public methods that 'choose' which one to use. 

We have the same variations in Scala, but they are exposed as trait mixins. I need to check how they end up on the client generation. My guess is they don't. At least not the mixins.